ui/src/app/zones/views/ZoneDetails/ZoneDetailsForm/ZoneDetailsForm.tsx

```diff
@@ -49,16 +53,19 @@ const ZoneForm = ({ id, closeForm }: Props): JSX.Element | null => {
         onSubmit={(values) => {
           dispatch(cleanup());
           dispatch(
-            zoneActions.update({
-              id: id,
-              description: values.description,
-              name: values.name,
-            })
+            zoneActions.update(
+              {
+                id: id,
+                description: values.description,
+                name: values.name,
+              },
+              formId.current
+            )
           );
         }}
         resetOnSave={true}
-        saved={saved}
-        saving={saving}
+        saved={updated}
+        saving={updating}
```

Comment:
As mentioned before, these should be related to the actual action being performed. `saved` and `saving` is a bit nebulous e.g. for a form that deletes the zone.